internal/pkg/manifest/env.go

```diff
@@ -102,9 +109,98 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID      *string              `yaml:"id,omitempty"`
-	CIDR    *IPNet               `yaml:"cidr,omitempty"`
-	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
+	ID                  *string              `yaml:"id,omitempty"`
+	CIDR                *IPNet               `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
+}
+
+type securityGroupConfig struct {
+	Ingress []SecurityGroupRule `yaml:"ingress,omitempty"`
+	Egress  []SecurityGroupRule `yaml:"egress,omitempty"`
+}
+
+func (cfg securityGroupConfig) isEmpty() bool {
+	return len(cfg.Ingress) == 0 && len(cfg.Egress) == 0
+}
+
+// SecurityGroupRule holds the security group ingress and egress configs
```

Comment:
nit: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences
```suggestion
// SecurityGroupRule holds the security group ingress and egress configs.
```

Reply:
🙈